apps/console/src/features/roles/pages/role-updated.tsx

```diff
@@ -0,0 +1,281 @@
+import { addAlert } from "@wso2is/core/store";
+import { ListLayout, PageLayout, PrimaryButton } from "@wso2is/react-components";
+import React, { FunctionComponent, ReactElement, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch } from "react-redux";
+import { RouteComponentProps } from "react-router";
+import { Dispatch } from "redux";
+import { Dropdown, DropdownItemProps, DropdownProps, Icon, PaginationProps } from "semantic-ui-react";
+import { AdvancedSearchWithBasicFilters, UIConstants } from "../../core";
+import { deleteRoleById, useRolesList } from "../api";
+import { RoleList } from "../components/role-list-updated";
+import { CreateRoleWizard } from "../components/wizard-updated/create-role-wizard";
+import { RoleConstants } from "../constants";
+
+type RolesPagePropsInterface = IdentifiableComponentInterface & RouteComponentProps;
+
+/**
+ * React component to list User Roles.
+ *
+ * @returns Roles page component.
+ */
+const RolesPage: FunctionComponent<RolesPagePropsInterface> = (
+    props: RolesPagePropsInterface
+): ReactElement => {
+    
+    const {
+        [ "data-componentid" ]: componentId
+    } = props;
+    
+    const dispatch: Dispatch = useDispatch();
+    const { t } = useTranslation();
+
+    const [ listItemLimit, setListItemLimit ] = useState<number>(UIConstants.DEFAULT_RESOURCE_LIST_ITEM_LIMIT);
+    const [ listOffset, setListOffset ] = useState<number>(0);
+    const [ showWizard, setShowWizard ] = useState<boolean>(false);
+    const [ filterBy, setFilterBy ] = useState<string>(undefined);
+    const [ isEmptyResults ] = useState<boolean>(false);
+    const [ triggerClearQuery, setTriggerClearQuery ] = useState<boolean>(false);
+
+    const {
+        data: rolesList,
+        isLoading: isRolesListLoading,
+        error: rolesListError,
+        mutate: mutateRolesList
+    } = useRolesList(
+        listItemLimit,
+        listOffset,
+        filterBy
+    );
+
+    /**
+     * Filter options for the roles list.
+     */
+    const filterOptions: DropdownItemProps[] = [
+        {
+            key: undefined,
+            text: t("console:manage.features.roles.list.filterOptions.all"),
+            value: undefined
+        },
+        {
+            key: RoleConstants.ROLE_AUDIENCE_APPLICATION_FILTER,
+            text: t("console:manage.features.roles.list.filterOptions.applicationRoles"),
+            value: RoleConstants.ROLE_AUDIENCE_APPLICATION_FILTER
+        },
+        {
+            key: RoleConstants.ROLE_AUDIENCE_ORGANIZATION_FILTER,
+            text: t("console:manage.features.roles.list.filterOptions.organizationRoles"),
+            value: RoleConstants.ROLE_AUDIENCE_ORGANIZATION_FILTER
+        }
+    ];
+
+    /**
+     * The following useEffect is used to handle if any error occurs while fetching the roles list.
+     */
+    useEffect(() => {
+        if (rolesListError) {
+            dispatch(addAlert<AlertInterface>({
```

Comment:
Shall we use `handleAlerts` function here as well? Since a separate function is defined.

Reply:
Fixed with https://github.com/wso2/identity-apps/pull/4280/commits/31b88ad29860876b65b8843de775b5b7a096cd5e